packages/gatsby/src/utils/api-runner-node.js

```diff
@@ -115,10 +115,128 @@ const deferActions = actions => {
   return deferred
 }
 
-const getLocalReporter = (activity, reporter) =>
-  activity
-    ? { ...reporter, panicOnBuild: activity.panicOnBuild.bind(activity) }
-    : reporter
+/**
+ * Create a local reporter
+ * Used to override reporter methods with activity methods
+ */
+function getLocalReporter({ activity, reporter }) {
+  // If we have an activity, bind panicOnBuild to the activities method to
+  // join them
+  if (activity) {
+    return { ...reporter, panicOnBuild: activity.panicOnBuild.bind(activity) }
+  }
+
+  return reporter
+}
+
+function extendErrorIdWithPluginName(pluginName, errorMeta) {
+  if (typeof errorMeta === `object`) {
+    const id = errorMeta && errorMeta[`id`]
+
+    if (id) {
+      errorMeta[`id`] = `${pluginName}_${id}`
+    }
+  }
+}
```

Comment:
Let's keep this side-effect free so we don't augment the object but instead create a new one.

```suggestion
function extendErrorIdWithPluginName(pluginName, errorMeta) {
  if (typeof errorMeta === `object`) {
    const id = errorMeta && errorMeta[`id`]

    if (id) {
      return {
        ...errorMeta,
        id: `${pluginName}_${id}`
      }
    }
  }
  
  return errorMeta;
}
```